Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/EMStudioSDK/Source/RenderPlugin/RenderUpdateCallback.cpp

```diff
@@ -273,5 +273,15 @@ namespace EMStudio
 
             renderUtil->RenderNodeNames(actorInstance, camera, screenWidth, screenHeight, renderOptions->GetNodeNameColor(), renderOptions->GetSelectedObjectColor(), visibleJointIndices, selectedJointIndices);
         }
+        else if (widget->GetRenderFlag(RenderViewWidget::RENDER_SELECTED_NODENAMES) && !selectedJointIndices.empty())
+        {
+            MCommon::Camera* camera = widget->GetRenderWidget()->GetCamera();
+            const uint32 screenWidth = widget->GetRenderWidget()->GetScreenWidth();
+            const uint32 screenHeight = widget->GetRenderWidget()->GetScreenHeight();
+
+            renderUtil->RenderNodeNames(
+                actorInstance, camera, screenWidth, screenHeight, renderOptions->GetNodeNameColor(),
+                renderOptions->GetSelectedObjectColor(), selectedJointIndices, selectedJointIndices);
+        }
```

Comment:
Can we just use the selection in the already present RENDER_NODENAMES option please? For the collider editor, the viewport already responds to the selection, so that behavior is already part of the editor. Adding a new option will just clutter the settings field.

Reply:
I'm not quite following you. Are you saying to just change the existing RENDER_NODENAMES command to do this behavior now? I left that as-is because it's still useful, it renders all the bone names, and the ones you have selected are a different color. This new command only renders the bones you have selected and none of the others.

Reply:
"I'm not quite following you. Are you saying to just change the existing RENDER_NODENAMES command to do this behavior now?" Yes. You can still render all bone names by either unselecting all joints or selecting them all. This behavior will then match the collider setup layout and UX's proposal. Elsewise we have different approaches in the tool for different layouts.